fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/api/client/rendereregistry/v1/ItemOverlayRenderer.java

```diff
@@ -0,0 +1,94 @@
+package net.fabricmc.fabric.api.client.rendereregistry.v1;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.render.Tessellator;
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.VertexFormats;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Matrix4f;
+import org.lwjgl.opengl.GL11;
+
+public interface ItemOverlayRenderer {
+	/**
+	 * @return <code>true</code> to cancel vanilla overlay rendering.
+	 */
+	boolean renderOverlay(MatrixStack matrixStack, float zOffset, TextRenderer renderer, ItemStack stack, int x, int y,
+						  String countLabel);
+
+	default void renderCountLabel(MatrixStack matrixStack, float zOffset, TextRenderer renderer, ItemStack stack,
```

Comment:
I am somewhat concerned about the pretty much raw copy code from vanilla that may change in the future

Reply:
What I was trying to go for here is not having to copy code from vanilla yourself, but yeah that's a valid concern.
Maybe I could have 3 separate "render overlay" methods for each part of the overlay...